rsconnect/actions.py

```diff
@@ -693,91 +700,63 @@ def deploy_jupyter_notebook(
     :return: the ultimate URL where the deployed app may be accessed and the sequence
     of log lines.  The log lines value will be None if a log callback was provided.
     """
-    app_store = AppStore(file_name)
-    (app_id, deployment_name, deployment_title, default_title, app_mode,) = gather_basic_deployment_info_for_notebook(
-        connect_server,
-        app_store,
-        file_name,
-        new,
-        app_id,
-        title,
-        static,
-    )
-    python, environment = get_python_env_info(
-        file_name,
-        python,
-        conda_mode=conda_mode,
-        force_generate=force_generate,
-    )
-    bundle = create_notebook_deployment_bundle(
-        file_name,
-        extra_files,
-        app_mode,
-        python,
-        environment,
-        True,
-        hide_all_input=hide_all_input,
-        hide_tagged_input=hide_tagged_input,
-        image=image,
-    )
-    return _finalize_deploy(
-        connect_server,
-        app_store,
-        file_name,
-        app_id,
-        app_mode,
-        deployment_name,
-        deployment_title,
-        default_title,
-        bundle,
-        log_callback,
-    )
+    kwargs = locals()
+    kwargs["extra_files"] = extra_files = validate_extra_files(dirname(file_name), extra_files)
+    app_mode = AppModes.JUPYTER_NOTEBOOK if not static else AppModes.STATIC
+
+    if isinstance(connect_server, api.RSConnectServer):
+        kwargs.update(
+            dict(
+                url=connect_server.url,
+                api_key=connect_server.api_key,
+                insecure=connect_server.insecure,
+                ca_data=connect_server.ca_data,
+                cookies=connect_server.cookie_jar,
+            )
+        )
+    elif isinstance(connect_server, api.ShinyappsServer) or isinstance(connect_server, api.CloudServer):
+        kwargs.update(
+            dict(
+                url=connect_server.url,
+                account=connect_server.account_name,
+                token=connect_server.token,
+                secret=connect_server.secret,
+            )
+        )
+    else:
+        raise RSConnectException("Unable to infer Connect client.")
 
+    base_dir = dirname(file_name)
+    _warn_on_ignored_manifest(base_dir)
+    _warn_if_no_requirements_file(base_dir)
+    _warn_if_environment_directory(base_dir)
```

Comment:
It seems a little odd for API usage to trigger console warnings. We should ask about whether that's a desired behavior.